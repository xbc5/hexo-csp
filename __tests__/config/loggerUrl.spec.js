"use strict";
const Config = require("../../lib/config");

function fixture() {
  return new Config({
    logger: {
      prod: { host: "https://prod-logger.com", path: "/custom-prod-logger" },
      dev: {
        host: "https://dev-logger.com",
        path: "/custom-dev-logger",
        port: 4000,
      },
    },
  });
}

["prod", "dev"].forEach((env) => {
  describe(`when given "${env}" as an arg`, () => {
    it(`should return "${env}" logger path`, async () => {
      const conf = fixture();
      expect(conf.loggerUrl(env)).toMatchSnapshot();
    });

    it(`should throw if value is undefined`, async () => {
      const conf = new Config();
      expect(() => conf.loggerUrl(env)).toThrow();
    });
  });
});
