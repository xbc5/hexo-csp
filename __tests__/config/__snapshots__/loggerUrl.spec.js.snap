// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when given "dev" as an arg should return "dev" logger path 1`] = `"https://dev-logger.com:4000/custom-dev-logger"`;

exports[`when given "prod" as an arg should return "prod" logger path 1`] = `"https://prod-logger.com/custom-prod-logger"`;
